// @linlak/ngxmomo/bundles/linlak-ngxmomo.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http'), require('rxjs')) :
    typeof define === 'function' && define.amd ? define('@linlak/ngxmomo', ['exports', '@angular/core', '@angular/common/http', 'rxjs'], factory) :
    (global = global || self, factory((global.linlak = global.linlak || {}, global.linlak.ngxmomo = {}), global.ng.core, global.ng.common.http, global.rxjs));
}(this, (function (exports, core, http, rxjs) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/entities/base-momo-provider.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @abstract
     */
    var /**
     * @abstract
     */
    BaseMomoProvider = /** @class */ (function () {
        function BaseMomoProvider() {
        }
        /**
         * @param {?} obj
         * @param {?} onload
         * @return {?}
         */
        BaseMomoProvider.prototype.loadScript = /**
         * @param {?} obj
         * @param {?} onload
         * @return {?}
         */
        function (obj, onload) {
            if (document.getElementById(obj.name)) {
                return;
            }
            /** @type {?} */
            var momoJs = document.createElement('script');
            momoJs.async = false;
            momoJs.src = obj.url;
            momoJs.onload = onload;
            document.head.appendChild(momoJs);
        };
        /**
         * @param {?} eventDetails
         * @return {?}
         */
        BaseMomoProvider.prototype.notify = /**
         * @param {?} eventDetails
         * @return {?}
         */
        function (eventDetails) {
            this.eventCallback({
                provider: this.TYPE,
                event: {
                    type: eventDetails.type,
                    details: eventDetails.detail
                }
            });
        };
        return BaseMomoProvider;
    }());
    if (false) {
        /** @type {?} */
        BaseMomoProvider.prototype.eventCallback;
        /** @type {?} */
        BaseMomoProvider.prototype.TYPE;
        /**
         * @abstract
         * @param {?} eventCallback
         * @return {?}
         */
        BaseMomoProvider.prototype.initialize = function (eventCallback) { };
        /**
         * @abstract
         * @return {?}
         */
        BaseMomoProvider.prototype.getApiUser = function () { };
        /**
         * @abstract
         * @return {?}
         */
        BaseMomoProvider.prototype.getCurrency = function () { };
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/entities/momo-provider.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @record
     */
    function MomoProvider() { }
    if (false) {
        /** @type {?} */
        MomoProvider.prototype.TYPE;
        /** @type {?} */
        MomoProvider.prototype.eventCallback;
        /**
         * @param {?} eventCallback
         * @return {?}
         */
        MomoProvider.prototype.initialize = function (eventCallback) { };
        /**
         * @param {?} obj
         * @param {?} onload
         * @return {?}
         */
        MomoProvider.prototype.loadScript = function (obj, onload) { };
        /**
         * @return {?}
         */
        MomoProvider.prototype.getApiUser = function () { };
        /**
         * @return {?}
         */
        MomoProvider.prototype.getCurrency = function () { };
        /**
         * @param {?} event
         * @return {?}
         */
        MomoProvider.prototype.notify = function (event) { };
    }
    /**
     * @record
     */
    function MtnMomoConfig() { }
    if (false) {
        /** @type {?} */
        MtnMomoConfig.prototype.apiUserId;
        /** @type {?|undefined} */
        MtnMomoConfig.prototype.environ;
        /** @type {?|undefined} */
        MtnMomoConfig.prototype.url;
        /** @type {?|undefined} */
        MtnMomoConfig.prototype.currency;
    }
    var MomoProviderClass = /** @class */ (function () {
        function MomoProviderClass() {
            this.currency = 'EUR';
            this.environ = 'sandbox';
        }
        return MomoProviderClass;
    }());
    if (false) {
        /** @type {?} */
        MomoProviderClass.prototype.name;
        /** @type {?} */
        MomoProviderClass.prototype.url;
        /** @type {?} */
        MomoProviderClass.prototype.currency;
        /** @type {?} */
        MomoProviderClass.prototype.environ;
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/providers/momo-widget-privider.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MomoWidgetProvider = /** @class */ (function (_super) {
        __extends(MomoWidgetProvider, _super);
        function MomoWidgetProvider(config) {
            var _this = _super.call(this) || this;
            _this.TYPE = 'mtnmomowidget';
            _this.momoProviderObj = new MomoProviderClass();
            _this.momoConfig = config;
            _this.momoProviderObj.name = 'mtnmomowidget';
            _this.momoProviderObj.url = config.url || 'https://widget.northeurope.cloudapp.azure.com:9443/v0.1.0/mobile-money-widget-mtn.js';
            return _this;
        }
        /**
         * @param {?} eventCallback
         * @return {?}
         */
        MomoWidgetProvider.prototype.initialize = /**
         * @param {?} eventCallback
         * @return {?}
         */
        function (eventCallback) {
            var _this = this;
            return new Promise((/**
             * @param {?} resolve
             * @param {?} reject
             * @return {?}
             */
            function (resolve, reject) {
                _this.loadScript(_this.momoProviderObj, (/**
                 * @return {?}
                 */
                function () {
                    _this.eventCallback = eventCallback;
                    window.addEventListener('mobile-money-qr-payment-created', (/**
                     * @param {?} e
                     * @return {?}
                     */
                    function (e) { return _this.notify(e); }));
                    window.addEventListener('mobile-money-qr-payment-canceled', (/**
                     * @param {?} e
                     * @return {?}
                     */
                    function (e) { return _this.notify(e); }));
                    window.addEventListener('mobile-money-qr-payment-successful', (/**
                     * @param {?} e
                     * @return {?}
                     */
                    function (e) { return _this.notify(e); }));
                    window.addEventListener('mobile-money-qr-payment-failed', (/**
                     * @param {?} e
                     * @return {?}
                     */
                    function (e) { return _this.notify(e); }));
                    mobileMoneyReinitializeWidgets();
                    resolve();
                }));
            }));
        };
        /**
         * @return {?}
         */
        MomoWidgetProvider.prototype.getApiUser = /**
         * @return {?}
         */
        function () {
            return this.momoConfig.apiUserId;
        };
        /**
         * @return {?}
         */
        MomoWidgetProvider.prototype.getCurrency = /**
         * @return {?}
         */
        function () {
            return this.momoConfig.currency;
        };
        MomoWidgetProvider.PROVIDER_TYPE = 'mtnmomowidget';
        return MomoWidgetProvider;
    }(BaseMomoProvider));
    if (false) {
        /** @type {?} */
        MomoWidgetProvider.PROVIDER_TYPE;
        /** @type {?} */
        MomoWidgetProvider.prototype.TYPE;
        /**
         * @type {?}
         * @private
         */
        MomoWidgetProvider.prototype.momoConfig;
        /** @type {?} */
        MomoWidgetProvider.prototype.momoProviderObj;
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/data/momo-cofig.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @record
     */
    function NgxMomoServiceConfigItem() { }
    if (false) {
        /** @type {?} */
        NgxMomoServiceConfigItem.prototype.provider;
    }
    var NgxMomoServiceConfig = /** @class */ (function () {
        function NgxMomoServiceConfig(providers) {
            this.providers = new Map();
            if (providers) {
                // tslint:disable-next-line: prefer-for-of
                for (var i = 0; i < providers.length; i++) {
                    /** @type {?} */
                    var element = providers[i];
                    this.providers.set(element.provider.TYPE, element.provider);
                }
            }
        }
        /**
         * @return {?}
         */
        NgxMomoServiceConfig.prototype.getProviders = /**
         * @return {?}
         */
        function () {
            return this.providers;
        };
        /**
         * @private
         * @param {?} provider
         * @return {?}
         */
        NgxMomoServiceConfig.prototype.addProvider = /**
         * @private
         * @param {?} provider
         * @return {?}
         */
        function (provider) {
            if (!this.providers.get(provider.TYPE)) {
                this.providers.set(provider.TYPE, provider);
            }
        };
        /**
         * @template THIS
         * @this {THIS}
         * @param {?} config
         * @return {THIS}
         */
        NgxMomoServiceConfig.prototype.addMomoWidgetProvider = /**
         * @template THIS
         * @this {THIS}
         * @param {?} config
         * @return {THIS}
         */
        function (config) {
            (/** @type {?} */ (this)).addProvider(new MomoWidgetProvider(config));
            return (/** @type {?} */ (this));
        };
        return NgxMomoServiceConfig;
    }());
    if (false) {
        /**
         * @type {?}
         * @private
         */
        NgxMomoServiceConfig.prototype.providers;
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/data/momo-events.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @record
     */
    function MomoEvent() { }
    if (false) {
        /** @type {?} */
        MomoEvent.prototype.type;
        /** @type {?} */
        MomoEvent.prototype.details;
    }
    /**
     * @record
     */
    function MomoEventMain() { }
    if (false) {
        /** @type {?} */
        MomoEventMain.prototype.provider;
        /** @type {?} */
        MomoEventMain.prototype.event;
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/data/index.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/ngxmomo.service.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxMomoService = /** @class */ (function () {
        function NgxMomoService(config, http) {
            var _this = this;
            this.http = http;
            this.eventListener = new rxjs.BehaviorSubject(false);
            this.providers = config.getProviders();
            this.providers.forEach((/**
             * @param {?} provider
             * @param {?} key
             * @return {?}
             */
            function (provider, key) {
                provider.initialize(_this.eventCallback.bind(_this)).then((/**
                 * @param {?} r
                 * @return {?}
                 */
                function (r) {
                }))
                    .catch((/**
                 * @param {?} error
                 * @return {?}
                 */
                function (error) {
                }));
            }));
        }
        /**
         * @return {?}
         */
        NgxMomoService.prototype.listen = /**
         * @return {?}
         */
        function () {
            return this.eventListener.asObservable();
        };
        /**
         * @private
         * @param {?} momoEvent
         * @return {?}
         */
        NgxMomoService.prototype.eventCallback = /**
         * @private
         * @param {?} momoEvent
         * @return {?}
         */
        function (momoEvent) {
            this.notifyEvt(momoEvent);
        };
        /**
         * @param {?} tagName
         * @return {?}
         */
        NgxMomoService.prototype.getProvider = /**
         * @param {?} tagName
         * @return {?}
         */
        function (tagName) {
            return this.providers.get(tagName);
        };
        /**
         * @private
         * @param {?} evt
         * @return {?}
         */
        NgxMomoService.prototype.notifyEvt = /**
         * @private
         * @param {?} evt
         * @return {?}
         */
        function (evt) {
            this.eventListener.next(evt);
        };
        NgxMomoService.decorators = [
            { type: core.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        NgxMomoService.ctorParameters = function () { return [
            { type: NgxMomoServiceConfig },
            { type: http.HttpClient }
        ]; };
        /** @nocollapse */ NgxMomoService.ngInjectableDef = core.ɵɵdefineInjectable({ factory: function NgxMomoService_Factory() { return new NgxMomoService(core.ɵɵinject(NgxMomoServiceConfig), core.ɵɵinject(http.HttpClient)); }, token: NgxMomoService, providedIn: "root" });
        return NgxMomoService;
    }());
    if (false) {
        /**
         * @type {?}
         * @private
         */
        NgxMomoService.prototype.providers;
        /** @type {?} */
        NgxMomoService.prototype.eventListener;
        /**
         * @type {?}
         * @private
         */
        NgxMomoService.prototype.http;
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/components/momo-qrpayment.component.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MomoQrpaymentComponent = /** @class */ (function () {
        function MomoQrpaymentComponent(momo) {
            this.momo = momo;
            this.momoProvider = this.momo.getProvider('mtnmomowidget');
            if (!this.momoProvider && !(this.momoProvider instanceof MomoWidgetProvider)) {
                throw new Error(MomoWidgetProvider.PROVIDER_TYPE + " was not initialized");
            }
            this.apiUserId = this.momoProvider.getApiUser();
            this.currency = this.momoProvider.getCurrency();
        }
        /**
         * @return {?}
         */
        MomoQrpaymentComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        /**
         * @return {?}
         */
        MomoQrpaymentComponent.prototype.ngOnDestroy = /**
         * @return {?}
         */
        function () {
        };
        MomoQrpaymentComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'momo-qrpayment',
                        template: "\n        <div>\n            <div\n                class=\"mobile-money-qr-payment\"\n                [attr.data-api-user-id]=\"apiUserId\"\n                [attr.data-amount]=\"amount\"\n                [attr.data-currency]=\"currency\"\n                [attr.data-external-id]=\"externalId\"\n            >\n            </div>\n        </div>\n    "
                    }] }
        ];
        /** @nocollapse */
        MomoQrpaymentComponent.ctorParameters = function () { return [
            { type: NgxMomoService }
        ]; };
        MomoQrpaymentComponent.propDecorators = {
            externalId: [{ type: core.Input }],
            amount: [{ type: core.Input }]
        };
        return MomoQrpaymentComponent;
    }());
    if (false) {
        /** @type {?} */
        MomoQrpaymentComponent.prototype.externalId;
        /** @type {?} */
        MomoQrpaymentComponent.prototype.amount;
        /**
         * @type {?}
         * @private
         */
        MomoQrpaymentComponent.prototype.momoProvider;
        /** @type {?} */
        MomoQrpaymentComponent.prototype.apiUserId;
        /** @type {?} */
        MomoQrpaymentComponent.prototype.currency;
        /** @type {?} */
        MomoQrpaymentComponent.prototype.momo;
    }

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/ngxmomo.module.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxMomoModule = /** @class */ (function () {
        function NgxMomoModule(parentModule) {
            if (parentModule) {
                NgxMomoModule.alreadCalled = true;
            }
        }
        /**
         * the forRoot function is used to initialize the momo library in the root module i.e AppModule
         * const momoConfig = new NgxMomoServiceConfig();
         *
         * import: [
         *  .............
         *  NgxMomoModule.forRoot(momoConfig),
         *  ..............
         * ]
         *
         */
        /**
         * the forRoot function is used to initialize the momo library in the root module i.e AppModule
         * const momoConfig = new NgxMomoServiceConfig();
         *
         * import: [
         *  .............
         *  NgxMomoModule.forRoot(momoConfig),
         *  ..............
         * ]
         *
         * @param {?} config
         * @return {?}
         */
        NgxMomoModule.forRoot = /**
         * the forRoot function is used to initialize the momo library in the root module i.e AppModule
         * const momoConfig = new NgxMomoServiceConfig();
         *
         * import: [
         *  .............
         *  NgxMomoModule.forRoot(momoConfig),
         *  ..............
         * ]
         *
         * @param {?} config
         * @return {?}
         */
        function (config) {
            if (NgxMomoModule.alreadCalled) {
                return;
            }
            return {
                ngModule: NgxMomoModule,
                providers: [
                    NgxMomoService,
                    {
                        provide: NgxMomoServiceConfig,
                        useValue: config
                    }
                ]
            };
        };
        NgxMomoModule.alreadCalled = false;
        NgxMomoModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [
                            MomoQrpaymentComponent,
                        ],
                        imports: [],
                        exports: [
                            MomoQrpaymentComponent,
                        ]
                    },] }
        ];
        /** @nocollapse */
        NgxMomoModule.ctorParameters = function () { return [
            { type: NgxMomoModule, decorators: [{ type: core.Optional }, { type: core.SkipSelf }] }
        ]; };
        return NgxMomoModule;
    }());
    if (false) {
        /**
         * @type {?}
         * @private
         */
        NgxMomoModule.alreadCalled;
    }

    exports.NgxMomoModule = NgxMomoModule;
    exports.NgxMomoService = NgxMomoService;
    exports.NgxMomoServiceConfig = NgxMomoServiceConfig;
    exports.ɵa = NgxMomoServiceConfig;
    exports.ɵb = MomoQrpaymentComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
