// @linlak/ngxmomo/esm2015/lib/ngxmomo.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngxmomo.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { NgxMomoServiceConfig } from './data';
import { HttpClient } from '@angular/common/http';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./data/momo-cofig";
import * as i2 from "@angular/common/http";
export class NgxMomoService {
    /**
     * @param {?} config
     * @param {?} http
     */
    constructor(config, http) {
        this.http = http;
        this.eventListener = new BehaviorSubject(false);
        this.providers = config.getProviders();
        this.providers.forEach((/**
         * @param {?} provider
         * @param {?} key
         * @return {?}
         */
        (provider, key) => {
            provider.initialize(this.eventCallback.bind(this)).then((/**
             * @param {?} r
             * @return {?}
             */
            r => {
            }))
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            error => {
            }));
        }));
    }
    /**
     * @return {?}
     */
    listen() {
        return this.eventListener.asObservable();
    }
    /**
     * @private
     * @param {?} momoEvent
     * @return {?}
     */
    eventCallback(momoEvent) {
        this.notifyEvt(momoEvent);
    }
    /**
     * @param {?} tagName
     * @return {?}
     */
    getProvider(tagName) {
        return this.providers.get(tagName);
    }
    /**
     * @private
     * @param {?} evt
     * @return {?}
     */
    notifyEvt(evt) {
        this.eventListener.next(evt);
    }
}
NgxMomoService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
NgxMomoService.ctorParameters = () => [
    { type: NgxMomoServiceConfig },
    { type: HttpClient }
];
/** @nocollapse */ NgxMomoService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NgxMomoService_Factory() { return new NgxMomoService(i0.ɵɵinject(i1.NgxMomoServiceConfig), i0.ɵɵinject(i2.HttpClient)); }, token: NgxMomoService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxMomoService.prototype.providers;
    /** @type {?} */
    NgxMomoService.prototype.eventListener;
    /**
     * @type {?}
     * @private
     */
    NgxMomoService.prototype.http;
}
