// @linlak/ngxmomo/esm2015/lib/ngxmomo.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngxmomo.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, Optional, SkipSelf } from '@angular/core';
import { NgxMomoServiceConfig } from './data';
import { NgxMomoService } from './ngxmomo.service';
import { MomoQrpaymentComponent } from './components/momo-qrpayment.component';
export class NgxMomoModule {
    /**
     * @param {?=} parentModule
     */
    constructor(parentModule) {
        if (parentModule) {
            NgxMomoModule.alreadCalled = true;
        }
    }
    /**
     * the forRoot function is used to initialize the momo library in the root module i.e AppModule
     * const momoConfig = new NgxMomoServiceConfig();
     *
     * import: [
     *  .............
     *  NgxMomoModule.forRoot(momoConfig),
     *  ..............
     * ]
     *
     * @param {?} config
     * @return {?}
     */
    static forRoot(config) {
        if (NgxMomoModule.alreadCalled) {
            return;
        }
        return {
            ngModule: NgxMomoModule,
            providers: [
                NgxMomoService,
                {
                    provide: NgxMomoServiceConfig,
                    useValue: config
                }
            ]
        };
    }
}
NgxMomoModule.alreadCalled = false;
NgxMomoModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    MomoQrpaymentComponent,
                ],
                imports: [],
                exports: [
                    MomoQrpaymentComponent,
                ]
            },] }
];
/** @nocollapse */
NgxMomoModule.ctorParameters = () => [
    { type: NgxMomoModule, decorators: [{ type: Optional }, { type: SkipSelf }] }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxMomoModule.alreadCalled;
}
