// @linlak/ngxmomo/fesm5/linlak-ngxmomo.js
import { __extends } from 'tslib';
import { Injectable, ɵɵdefineInjectable, ɵɵinject, Component, Input, NgModule, Optional, SkipSelf } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { BehaviorSubject } from 'rxjs';

/**
 * @fileoverview added by tsickle
 * Generated from: lib/entities/base-momo-provider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @abstract
 */
var /**
 * @abstract
 */
BaseMomoProvider = /** @class */ (function () {
    function BaseMomoProvider() {
    }
    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    BaseMomoProvider.prototype.loadScript = /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    function (obj, onload) {
        if (document.getElementById(obj.name)) {
            return;
        }
        /** @type {?} */
        var momoJs = document.createElement('script');
        momoJs.async = false;
        momoJs.src = obj.url;
        momoJs.onload = onload;
        document.head.appendChild(momoJs);
    };
    /**
     * @param {?} eventDetails
     * @return {?}
     */
    BaseMomoProvider.prototype.notify = /**
     * @param {?} eventDetails
     * @return {?}
     */
    function (eventDetails) {
        this.eventCallback({
            provider: this.TYPE,
            event: {
                type: eventDetails.type,
                details: eventDetails.detail
            }
        });
    };
    return BaseMomoProvider;
}());
if (false) {
    /** @type {?} */
    BaseMomoProvider.prototype.eventCallback;
    /** @type {?} */
    BaseMomoProvider.prototype.TYPE;
    /**
     * @abstract
     * @param {?} eventCallback
     * @return {?}
     */
    BaseMomoProvider.prototype.initialize = function (eventCallback) { };
    /**
     * @abstract
     * @return {?}
     */
    BaseMomoProvider.prototype.getApiUser = function () { };
    /**
     * @abstract
     * @return {?}
     */
    BaseMomoProvider.prototype.getCurrency = function () { };
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/entities/momo-provider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
function MomoProvider() { }
if (false) {
    /** @type {?} */
    MomoProvider.prototype.TYPE;
    /** @type {?} */
    MomoProvider.prototype.eventCallback;
    /**
     * @param {?} eventCallback
     * @return {?}
     */
    MomoProvider.prototype.initialize = function (eventCallback) { };
    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    MomoProvider.prototype.loadScript = function (obj, onload) { };
    /**
     * @return {?}
     */
    MomoProvider.prototype.getApiUser = function () { };
    /**
     * @return {?}
     */
    MomoProvider.prototype.getCurrency = function () { };
    /**
     * @param {?} event
     * @return {?}
     */
    MomoProvider.prototype.notify = function (event) { };
}
/**
 * @record
 */
function MtnMomoConfig() { }
if (false) {
    /** @type {?} */
    MtnMomoConfig.prototype.apiUserId;
    /** @type {?|undefined} */
    MtnMomoConfig.prototype.environ;
    /** @type {?|undefined} */
    MtnMomoConfig.prototype.url;
    /** @type {?|undefined} */
    MtnMomoConfig.prototype.currency;
}
var MomoProviderClass = /** @class */ (function () {
    function MomoProviderClass() {
        this.currency = 'EUR';
        this.environ = 'sandbox';
    }
    return MomoProviderClass;
}());
if (false) {
    /** @type {?} */
    MomoProviderClass.prototype.name;
    /** @type {?} */
    MomoProviderClass.prototype.url;
    /** @type {?} */
    MomoProviderClass.prototype.currency;
    /** @type {?} */
    MomoProviderClass.prototype.environ;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/providers/momo-widget-privider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MomoWidgetProvider = /** @class */ (function (_super) {
    __extends(MomoWidgetProvider, _super);
    function MomoWidgetProvider(config) {
        var _this = _super.call(this) || this;
        _this.TYPE = 'mtnmomowidget';
        _this.momoProviderObj = new MomoProviderClass();
        _this.momoConfig = config;
        _this.momoProviderObj.name = 'mtnmomowidget';
        _this.momoProviderObj.url = config.url || 'https://widget.northeurope.cloudapp.azure.com:9443/v0.1.0/mobile-money-widget-mtn.js';
        return _this;
    }
    /**
     * @param {?} eventCallback
     * @return {?}
     */
    MomoWidgetProvider.prototype.initialize = /**
     * @param {?} eventCallback
     * @return {?}
     */
    function (eventCallback) {
        var _this = this;
        return new Promise((/**
         * @param {?} resolve
         * @param {?} reject
         * @return {?}
         */
        function (resolve, reject) {
            _this.loadScript(_this.momoProviderObj, (/**
             * @return {?}
             */
            function () {
                _this.eventCallback = eventCallback;
                window.addEventListener('mobile-money-qr-payment-created', (/**
                 * @param {?} e
                 * @return {?}
                 */
                function (e) { return _this.notify(e); }));
                window.addEventListener('mobile-money-qr-payment-canceled', (/**
                 * @param {?} e
                 * @return {?}
                 */
                function (e) { return _this.notify(e); }));
                window.addEventListener('mobile-money-qr-payment-successful', (/**
                 * @param {?} e
                 * @return {?}
                 */
                function (e) { return _this.notify(e); }));
                window.addEventListener('mobile-money-qr-payment-failed', (/**
                 * @param {?} e
                 * @return {?}
                 */
                function (e) { return _this.notify(e); }));
                mobileMoneyReinitializeWidgets();
                resolve();
            }));
        }));
    };
    /**
     * @return {?}
     */
    MomoWidgetProvider.prototype.getApiUser = /**
     * @return {?}
     */
    function () {
        return this.momoConfig.apiUserId;
    };
    /**
     * @return {?}
     */
    MomoWidgetProvider.prototype.getCurrency = /**
     * @return {?}
     */
    function () {
        return this.momoConfig.currency;
    };
    MomoWidgetProvider.PROVIDER_TYPE = 'mtnmomowidget';
    return MomoWidgetProvider;
}(BaseMomoProvider));
if (false) {
    /** @type {?} */
    MomoWidgetProvider.PROVIDER_TYPE;
    /** @type {?} */
    MomoWidgetProvider.prototype.TYPE;
    /**
     * @type {?}
     * @private
     */
    MomoWidgetProvider.prototype.momoConfig;
    /** @type {?} */
    MomoWidgetProvider.prototype.momoProviderObj;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/data/momo-cofig.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
function NgxMomoServiceConfigItem() { }
if (false) {
    /** @type {?} */
    NgxMomoServiceConfigItem.prototype.provider;
}
var NgxMomoServiceConfig = /** @class */ (function () {
    function NgxMomoServiceConfig(providers) {
        this.providers = new Map();
        if (providers) {
            // tslint:disable-next-line: prefer-for-of
            for (var i = 0; i < providers.length; i++) {
                /** @type {?} */
                var element = providers[i];
                this.providers.set(element.provider.TYPE, element.provider);
            }
        }
    }
    /**
     * @return {?}
     */
    NgxMomoServiceConfig.prototype.getProviders = /**
     * @return {?}
     */
    function () {
        return this.providers;
    };
    /**
     * @private
     * @param {?} provider
     * @return {?}
     */
    NgxMomoServiceConfig.prototype.addProvider = /**
     * @private
     * @param {?} provider
     * @return {?}
     */
    function (provider) {
        if (!this.providers.get(provider.TYPE)) {
            this.providers.set(provider.TYPE, provider);
        }
    };
    /**
     * @template THIS
     * @this {THIS}
     * @param {?} config
     * @return {THIS}
     */
    NgxMomoServiceConfig.prototype.addMomoWidgetProvider = /**
     * @template THIS
     * @this {THIS}
     * @param {?} config
     * @return {THIS}
     */
    function (config) {
        (/** @type {?} */ (this)).addProvider(new MomoWidgetProvider(config));
        return (/** @type {?} */ (this));
    };
    return NgxMomoServiceConfig;
}());
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxMomoServiceConfig.prototype.providers;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/data/momo-events.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
function MomoEvent() { }
if (false) {
    /** @type {?} */
    MomoEvent.prototype.type;
    /** @type {?} */
    MomoEvent.prototype.details;
}
/**
 * @record
 */
function MomoEventMain() { }
if (false) {
    /** @type {?} */
    MomoEventMain.prototype.provider;
    /** @type {?} */
    MomoEventMain.prototype.event;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/data/index.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngxmomo.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxMomoService = /** @class */ (function () {
    function NgxMomoService(config, http) {
        var _this = this;
        this.http = http;
        this.eventListener = new BehaviorSubject(false);
        this.providers = config.getProviders();
        this.providers.forEach((/**
         * @param {?} provider
         * @param {?} key
         * @return {?}
         */
        function (provider, key) {
            provider.initialize(_this.eventCallback.bind(_this)).then((/**
             * @param {?} r
             * @return {?}
             */
            function (r) {
            }))
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            function (error) {
            }));
        }));
    }
    /**
     * @return {?}
     */
    NgxMomoService.prototype.listen = /**
     * @return {?}
     */
    function () {
        return this.eventListener.asObservable();
    };
    /**
     * @private
     * @param {?} momoEvent
     * @return {?}
     */
    NgxMomoService.prototype.eventCallback = /**
     * @private
     * @param {?} momoEvent
     * @return {?}
     */
    function (momoEvent) {
        this.notifyEvt(momoEvent);
    };
    /**
     * @param {?} tagName
     * @return {?}
     */
    NgxMomoService.prototype.getProvider = /**
     * @param {?} tagName
     * @return {?}
     */
    function (tagName) {
        return this.providers.get(tagName);
    };
    /**
     * @private
     * @param {?} evt
     * @return {?}
     */
    NgxMomoService.prototype.notifyEvt = /**
     * @private
     * @param {?} evt
     * @return {?}
     */
    function (evt) {
        this.eventListener.next(evt);
    };
    NgxMomoService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NgxMomoService.ctorParameters = function () { return [
        { type: NgxMomoServiceConfig },
        { type: HttpClient }
    ]; };
    /** @nocollapse */ NgxMomoService.ngInjectableDef = ɵɵdefineInjectable({ factory: function NgxMomoService_Factory() { return new NgxMomoService(ɵɵinject(NgxMomoServiceConfig), ɵɵinject(HttpClient)); }, token: NgxMomoService, providedIn: "root" });
    return NgxMomoService;
}());
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxMomoService.prototype.providers;
    /** @type {?} */
    NgxMomoService.prototype.eventListener;
    /**
     * @type {?}
     * @private
     */
    NgxMomoService.prototype.http;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/momo-qrpayment.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var MomoQrpaymentComponent = /** @class */ (function () {
    function MomoQrpaymentComponent(momo) {
        this.momo = momo;
        this.momoProvider = this.momo.getProvider('mtnmomowidget');
        if (!this.momoProvider && !(this.momoProvider instanceof MomoWidgetProvider)) {
            throw new Error(MomoWidgetProvider.PROVIDER_TYPE + " was not initialized");
        }
        this.apiUserId = this.momoProvider.getApiUser();
        this.currency = this.momoProvider.getCurrency();
    }
    /**
     * @return {?}
     */
    MomoQrpaymentComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    MomoQrpaymentComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
    };
    MomoQrpaymentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'momo-qrpayment',
                    template: "\n        <div>\n            <div\n                class=\"mobile-money-qr-payment\"\n                [attr.data-api-user-id]=\"apiUserId\"\n                [attr.data-amount]=\"amount\"\n                [attr.data-currency]=\"currency\"\n                [attr.data-external-id]=\"externalId\"\n            >\n            </div>\n        </div>\n    "
                }] }
    ];
    /** @nocollapse */
    MomoQrpaymentComponent.ctorParameters = function () { return [
        { type: NgxMomoService }
    ]; };
    MomoQrpaymentComponent.propDecorators = {
        externalId: [{ type: Input }],
        amount: [{ type: Input }]
    };
    return MomoQrpaymentComponent;
}());
if (false) {
    /** @type {?} */
    MomoQrpaymentComponent.prototype.externalId;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.amount;
    /**
     * @type {?}
     * @private
     */
    MomoQrpaymentComponent.prototype.momoProvider;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.apiUserId;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.currency;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.momo;
}

/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngxmomo.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxMomoModule = /** @class */ (function () {
    function NgxMomoModule(parentModule) {
        if (parentModule) {
            NgxMomoModule.alreadCalled = true;
        }
    }
    /**
     * the forRoot function is used to initialize the momo library in the root module i.e AppModule
     * const momoConfig = new NgxMomoServiceConfig();
     *
     * import: [
     *  .............
     *  NgxMomoModule.forRoot(momoConfig),
     *  ..............
     * ]
     *
     */
    /**
     * the forRoot function is used to initialize the momo library in the root module i.e AppModule
     * const momoConfig = new NgxMomoServiceConfig();
     *
     * import: [
     *  .............
     *  NgxMomoModule.forRoot(momoConfig),
     *  ..............
     * ]
     *
     * @param {?} config
     * @return {?}
     */
    NgxMomoModule.forRoot = /**
     * the forRoot function is used to initialize the momo library in the root module i.e AppModule
     * const momoConfig = new NgxMomoServiceConfig();
     *
     * import: [
     *  .............
     *  NgxMomoModule.forRoot(momoConfig),
     *  ..............
     * ]
     *
     * @param {?} config
     * @return {?}
     */
    function (config) {
        if (NgxMomoModule.alreadCalled) {
            return;
        }
        return {
            ngModule: NgxMomoModule,
            providers: [
                NgxMomoService,
                {
                    provide: NgxMomoServiceConfig,
                    useValue: config
                }
            ]
        };
    };
    NgxMomoModule.alreadCalled = false;
    NgxMomoModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        MomoQrpaymentComponent,
                    ],
                    imports: [],
                    exports: [
                        MomoQrpaymentComponent,
                    ]
                },] }
    ];
    /** @nocollapse */
    NgxMomoModule.ctorParameters = function () { return [
        { type: NgxMomoModule, decorators: [{ type: Optional }, { type: SkipSelf }] }
    ]; };
    return NgxMomoModule;
}());
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxMomoModule.alreadCalled;
}

/**
 * @fileoverview added by tsickle
 * Generated from: index.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: public-api.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: linlak-ngxmomo.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NgxMomoModule, NgxMomoService, NgxMomoServiceConfig, NgxMomoServiceConfig as ɵa, MomoQrpaymentComponent as ɵb };
