// @linlak/ngxmomo/esm5/lib/components/momo-qrpayment.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/momo-qrpayment.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { NgxMomoService } from '../ngxmomo.service';
import { MomoWidgetProvider } from '../providers/momo-widget-privider';
var MomoQrpaymentComponent = /** @class */ (function () {
    function MomoQrpaymentComponent(momo) {
        this.momo = momo;
        this.momoProvider = this.momo.getProvider('mtnmomowidget');
        if (!this.momoProvider && !(this.momoProvider instanceof MomoWidgetProvider)) {
            throw new Error(MomoWidgetProvider.PROVIDER_TYPE + " was not initialized");
        }
        this.apiUserId = this.momoProvider.getApiUser();
        this.currency = this.momoProvider.getCurrency();
    }
    /**
     * @return {?}
     */
    MomoQrpaymentComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    MomoQrpaymentComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
    };
    MomoQrpaymentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'momo-qrpayment',
                    template: "\n        <div>\n            <div\n                class=\"mobile-money-qr-payment\"\n                [attr.data-api-user-id]=\"apiUserId\"\n                [attr.data-amount]=\"amount\"\n                [attr.data-currency]=\"currency\"\n                [attr.data-external-id]=\"externalId\"\n            >\n            </div>\n        </div>\n    "
                }] }
    ];
    /** @nocollapse */
    MomoQrpaymentComponent.ctorParameters = function () { return [
        { type: NgxMomoService }
    ]; };
    MomoQrpaymentComponent.propDecorators = {
        externalId: [{ type: Input }],
        amount: [{ type: Input }]
    };
    return MomoQrpaymentComponent;
}());
export { MomoQrpaymentComponent };
if (false) {
    /** @type {?} */
    MomoQrpaymentComponent.prototype.externalId;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.amount;
    /**
     * @type {?}
     * @private
     */
    MomoQrpaymentComponent.prototype.momoProvider;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.apiUserId;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.currency;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.momo;
}
