// @linlak/ngxmomo/esm5/lib/ngxmomo.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngxmomo.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { NgxMomoServiceConfig } from './data';
import { HttpClient } from '@angular/common/http';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./data/momo-cofig";
import * as i2 from "@angular/common/http";
var NgxMomoService = /** @class */ (function () {
    function NgxMomoService(config, http) {
        var _this = this;
        this.http = http;
        this.eventListener = new BehaviorSubject(false);
        this.providers = config.getProviders();
        this.providers.forEach((/**
         * @param {?} provider
         * @param {?} key
         * @return {?}
         */
        function (provider, key) {
            provider.initialize(_this.eventCallback.bind(_this)).then((/**
             * @param {?} r
             * @return {?}
             */
            function (r) {
            }))
                .catch((/**
             * @param {?} error
             * @return {?}
             */
            function (error) {
            }));
        }));
    }
    /**
     * @return {?}
     */
    NgxMomoService.prototype.listen = /**
     * @return {?}
     */
    function () {
        return this.eventListener.asObservable();
    };
    /**
     * @private
     * @param {?} momoEvent
     * @return {?}
     */
    NgxMomoService.prototype.eventCallback = /**
     * @private
     * @param {?} momoEvent
     * @return {?}
     */
    function (momoEvent) {
        this.notifyEvt(momoEvent);
    };
    /**
     * @param {?} tagName
     * @return {?}
     */
    NgxMomoService.prototype.getProvider = /**
     * @param {?} tagName
     * @return {?}
     */
    function (tagName) {
        return this.providers.get(tagName);
    };
    /**
     * @private
     * @param {?} evt
     * @return {?}
     */
    NgxMomoService.prototype.notifyEvt = /**
     * @private
     * @param {?} evt
     * @return {?}
     */
    function (evt) {
        this.eventListener.next(evt);
    };
    NgxMomoService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NgxMomoService.ctorParameters = function () { return [
        { type: NgxMomoServiceConfig },
        { type: HttpClient }
    ]; };
    /** @nocollapse */ NgxMomoService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NgxMomoService_Factory() { return new NgxMomoService(i0.ɵɵinject(i1.NgxMomoServiceConfig), i0.ɵɵinject(i2.HttpClient)); }, token: NgxMomoService, providedIn: "root" });
    return NgxMomoService;
}());
export { NgxMomoService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxMomoService.prototype.providers;
    /** @type {?} */
    NgxMomoService.prototype.eventListener;
    /**
     * @type {?}
     * @private
     */
    NgxMomoService.prototype.http;
}
