// @linlak/ngxmomo/esm5/lib/entities/base-momo-provider.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/entities/base-momo-provider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @abstract
 */
var /**
 * @abstract
 */
BaseMomoProvider = /** @class */ (function () {
    function BaseMomoProvider() {
    }
    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    BaseMomoProvider.prototype.loadScript = /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    function (obj, onload) {
        if (document.getElementById(obj.name)) {
            return;
        }
        /** @type {?} */
        var momoJs = document.createElement('script');
        momoJs.async = false;
        momoJs.src = obj.url;
        momoJs.onload = onload;
        document.head.appendChild(momoJs);
    };
    /**
     * @param {?} eventDetails
     * @return {?}
     */
    BaseMomoProvider.prototype.notify = /**
     * @param {?} eventDetails
     * @return {?}
     */
    function (eventDetails) {
        this.eventCallback({
            provider: this.TYPE,
            event: {
                type: eventDetails.type,
                details: eventDetails.detail
            }
        });
    };
    return BaseMomoProvider;
}());
/**
 * @abstract
 */
export { BaseMomoProvider };
if (false) {
    /** @type {?} */
    BaseMomoProvider.prototype.eventCallback;
    /** @type {?} */
    BaseMomoProvider.prototype.TYPE;
    /**
     * @abstract
     * @param {?} eventCallback
     * @return {?}
     */
    BaseMomoProvider.prototype.initialize = function (eventCallback) { };
    /**
     * @abstract
     * @return {?}
     */
    BaseMomoProvider.prototype.getApiUser = function () { };
    /**
     * @abstract
     * @return {?}
     */
    BaseMomoProvider.prototype.getCurrency = function () { };
}
