// @linlak/ngxmomo/esm2015/lib/providers/momo-widget-privider.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/providers/momo-widget-privider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { BaseMomoProvider } from '../entities/base-momo-provider';
import { MomoProviderClass } from '../entities/momo-provider';
export class MomoWidgetProvider extends BaseMomoProvider {
    /**
     * @param {?} config
     */
    constructor(config) {
        super();
        this.TYPE = 'mtnmomowidget';
        this.momoProviderObj = new MomoProviderClass();
        this.momoConfig = config;
        this.momoProviderObj.name = 'mtnmomowidget';
        this.momoProviderObj.url = config.url || 'https://widget.northeurope.cloudapp.azure.com:9443/v0.1.0/mobile-money-widget-mtn.js';
    }
    /**
     * @param {?} eventCallback
     * @return {?}
     */
    initialize(eventCallback) {
        return new Promise((/**
         * @param {?} resolve
         * @param {?} reject
         * @return {?}
         */
        (resolve, reject) => {
            this.loadScript(this.momoProviderObj, (/**
             * @return {?}
             */
            () => {
                this.eventCallback = eventCallback;
                window.addEventListener('mobile-money-qr-payment-created', (/**
                 * @param {?} e
                 * @return {?}
                 */
                (e) => this.notify(e)));
                window.addEventListener('mobile-money-qr-payment-canceled', (/**
                 * @param {?} e
                 * @return {?}
                 */
                (e) => this.notify(e)));
                window.addEventListener('mobile-money-qr-payment-successful', (/**
                 * @param {?} e
                 * @return {?}
                 */
                (e) => this.notify(e)));
                window.addEventListener('mobile-money-qr-payment-failed', (/**
                 * @param {?} e
                 * @return {?}
                 */
                (e) => this.notify(e)));
                mobileMoneyReinitializeWidgets();
                resolve();
            }));
        }));
    }
    /**
     * @return {?}
     */
    getApiUser() {
        return this.momoConfig.apiUserId;
    }
    /**
     * @return {?}
     */
    getCurrency() {
        return this.momoConfig.currency;
    }
}
MomoWidgetProvider.PROVIDER_TYPE = 'mtnmomowidget';
if (false) {
    /** @type {?} */
    MomoWidgetProvider.PROVIDER_TYPE;
    /** @type {?} */
    MomoWidgetProvider.prototype.TYPE;
    /**
     * @type {?}
     * @private
     */
    MomoWidgetProvider.prototype.momoConfig;
    /** @type {?} */
    MomoWidgetProvider.prototype.momoProviderObj;
}
