// @linlak/ngxmomo/esm2015/lib/components/momo-qrpayment.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/components/momo-qrpayment.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { NgxMomoService } from '../ngxmomo.service';
import { MomoWidgetProvider } from '../providers/momo-widget-privider';
export class MomoQrpaymentComponent {
    /**
     * @param {?} momo
     */
    constructor(momo) {
        this.momo = momo;
        this.momoProvider = this.momo.getProvider('mtnmomowidget');
        if (!this.momoProvider && !(this.momoProvider instanceof MomoWidgetProvider)) {
            throw new Error(`${MomoWidgetProvider.PROVIDER_TYPE} was not initialized`);
        }
        this.apiUserId = this.momoProvider.getApiUser();
        this.currency = this.momoProvider.getCurrency();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
    }
}
MomoQrpaymentComponent.decorators = [
    { type: Component, args: [{
                selector: 'momo-qrpayment',
                template: `
        <div>
            <div
                class="mobile-money-qr-payment"
                [attr.data-api-user-id]="apiUserId"
                [attr.data-amount]="amount"
                [attr.data-currency]="currency"
                [attr.data-external-id]="externalId"
            >
            </div>
        </div>
    `
            }] }
];
/** @nocollapse */
MomoQrpaymentComponent.ctorParameters = () => [
    { type: NgxMomoService }
];
MomoQrpaymentComponent.propDecorators = {
    externalId: [{ type: Input }],
    amount: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    MomoQrpaymentComponent.prototype.externalId;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.amount;
    /**
     * @type {?}
     * @private
     */
    MomoQrpaymentComponent.prototype.momoProvider;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.apiUserId;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.currency;
    /** @type {?} */
    MomoQrpaymentComponent.prototype.momo;
}
