// @linlak/ngxmomo/esm2015/lib/entities/base-momo-provider.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/entities/base-momo-provider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @abstract
 */
export class BaseMomoProvider {
    constructor() { }
    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    loadScript(obj, onload) {
        if (document.getElementById(obj.name)) {
            return;
        }
        /** @type {?} */
        const momoJs = document.createElement('script');
        momoJs.async = false;
        momoJs.src = obj.url;
        momoJs.onload = onload;
        document.head.appendChild(momoJs);
    }
    /**
     * @param {?} eventDetails
     * @return {?}
     */
    notify(eventDetails) {
        this.eventCallback({
            provider: this.TYPE,
            event: {
                type: eventDetails.type,
                details: eventDetails.detail
            }
        });
    }
}
if (false) {
    /** @type {?} */
    BaseMomoProvider.prototype.eventCallback;
    /** @type {?} */
    BaseMomoProvider.prototype.TYPE;
    /**
     * @abstract
     * @param {?} eventCallback
     * @return {?}
     */
    BaseMomoProvider.prototype.initialize = function (eventCallback) { };
    /**
     * @abstract
     * @return {?}
     */
    BaseMomoProvider.prototype.getApiUser = function () { };
    /**
     * @abstract
     * @return {?}
     */
    BaseMomoProvider.prototype.getCurrency = function () { };
}
