// @linlak/ngxmomo/esm2015/lib/entities/momo-provider.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/entities/momo-provider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function MomoProvider() { }
if (false) {
    /** @type {?} */
    MomoProvider.prototype.TYPE;
    /** @type {?} */
    MomoProvider.prototype.eventCallback;
    /**
     * @param {?} eventCallback
     * @return {?}
     */
    MomoProvider.prototype.initialize = function (eventCallback) { };
    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    MomoProvider.prototype.loadScript = function (obj, onload) { };
    /**
     * @return {?}
     */
    MomoProvider.prototype.getApiUser = function () { };
    /**
     * @return {?}
     */
    MomoProvider.prototype.getCurrency = function () { };
    /**
     * @param {?} event
     * @return {?}
     */
    MomoProvider.prototype.notify = function (event) { };
}
/**
 * @record
 */
export function MtnMomoConfig() { }
if (false) {
    /** @type {?} */
    MtnMomoConfig.prototype.apiUserId;
    /** @type {?|undefined} */
    MtnMomoConfig.prototype.environ;
    /** @type {?|undefined} */
    MtnMomoConfig.prototype.url;
    /** @type {?|undefined} */
    MtnMomoConfig.prototype.currency;
}
export class MomoProviderClass {
    constructor() {
        this.currency = 'EUR';
        this.environ = 'sandbox';
    }
}
if (false) {
    /** @type {?} */
    MomoProviderClass.prototype.name;
    /** @type {?} */
    MomoProviderClass.prototype.url;
    /** @type {?} */
    MomoProviderClass.prototype.currency;
    /** @type {?} */
    MomoProviderClass.prototype.environ;
}
